### PR TITLE
feat(cli): clean up log level features

### LINKED ISSUE #3403: Clean up log level features
### Describe the feature

The CLI has a feature named `only-info-logs` which enables `tracing/release_max_level_info`.

https://github.com/paradigmxyz/reth/blob/23a99d41643d3a432cae0f717c90451738ec2d51/bin/reth/Cargo.toml#L88-L91

- The feature should be renamed to something like `min-info-logs`
- We should have one for each log level, since the current one hides warnings and errors (not good)
- If the feature is referred to anywhere (e.g. the book), it should be updated
- If we are currently using `only-info-logs`, we should use the one that corresponds to warning + error + info

See here https://docs.rs/tracing/latest/tracing/level_filters/index.html#compile-time-filters

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by tcoratger**
Log level features modification to include:
```shell
release_max_level_off
release_max_level_error
release_max_level_warn
release_max_level_info
release_max_level_debug
release_max_level_trace
```
This PR also modifies the current usage to take into account warning + error + info for a better log coverage.

Should solve #3403.

### CODE REVIEW

File: bin/reth/src/lib.rs [hunk 1]
```diff
@@ -16,8 +16,8 @@
 //!   and leak detection functionality. See [jemalloc's opt.prof](https://jemalloc.net/jemalloc.3.html#opt.prof)
 //!   documentation for usage details. This is **not recommended on Windows**. See [here](https://rust-lang.github.io/rfcs/1974-global-allocators.html#jemalloc)
 //!   for more info.
-//! - `only-info-logs`: Disables all logs below `info` level. This can speed up the node, since
-//!   fewer calls to the logging component is made.
+//! - `min-info-logs`: Disables all logs below `info` level. This can speed up the node, since fewer
```
Comment (onbjerg) [inline bin/reth/src/lib.rs:19]: please add the remaining features here just like this one so it is documented

**Review by onbjerg — CHANGES_REQUESTED** [review]
ty for this, see my comment

File: bin/reth/src/lib.rs (hunk 1 above)
Comment (tcoratger) [reply bin/reth/src/lib.rs:21]: @onbjerg Done

**Review by onbjerg — APPROVED** [review]
tyty

**Pull request merged**